### DISCUSSION TITLE
How to install from source?

### DISCUSSION THREAD

**Discussion by marathone** [ihabunek/toot#503] [Q&A] [answered]
The latest version of toot, doesn't seem to install the same way as previous versions. Can someone please give succinct instructions on how to install from source on Debian Stable? Thank-you!

**Comment by ihabunek** [comment, marked as answer]
toot recently migrated from setup.py to pyproject.toml, so i'm guessing that's your problem.

To install toot in a virtual environment, so it doesn't pollute your global/user python env you can:

```
git clone git@github.com:ihabunek/toot.git
cd toot
python3 -m venv .venv
source .venv/bin/activate
pip install . 
```

After this, the executable is available at `.venv/bin/toot`.

Does this work for you?

**Reply by marathone** [reply]
Thanks for the quick response - shouldn't this have been documented in the README's? I decided to download using the python repository. It says the version is 44.1 yet the webpage says the latest is 43.0 or did yesterday. Is this a legit version? Thanks.

**Reply by ihabunek** [reply]
Installing python is a pain and everybody has their own workflow. I would usually not recommend users to install the way described above, but rather from a) distro package, b) python package index. I'm also seriously considering putting a third party pakcage manager like pipx or uv as the recommended method as it just works...

I agree the docs could use some love.

**Reply by ihabunek** [reply]
Also the web site is not currently deployed automatically, I've updated it now. Official versions are the ones you find when running `git tag`.

**Comment by ihabunek** [comment]
Updated installation instructions
https://toot.bezdomni.net/installation.html
